tests/admin_views/tests.py

```diff
@@ -5132,6 +5135,44 @@ def test_readonly_foreignkey_links(self):
         # Related ForeignKey object not registered in admin.
         self.assertContains(response, '<div class="readonly">Chapter 1</div>', html=True)
 
+    def test_readonly_foreignkey_links_custom_adminsite(self):
+        """
+        Alternate admin site generates proper links for readonly fields
+        """
+        chapter = Chapter.objects.create(
+            title='Chapter 1',
+            content='content',
+            book=Book.objects.create(name='Book 1'),
+        )
+        language = Language.objects.create(iso='_40', name='Test')
+        obj = ReadOnlyRelatedField.objects.create(
+            chapter=chapter,
+            language=language,
+            user=self.superuser,
+        )
+
+        request = RequestFactory().get(
+            reverse(
+                'namespaced_admin:admin_views_readonlyrelatedfield_change',
+                args=(obj.pk,)
+            )
+        )
+
+        # Get an AdminReadonlyField for the alternate admin site
+        related_admin = site2._registry[ReadOnlyRelatedField]
+        admin_form = related_admin.get_form(request, obj=obj)()
+        readonly_field = AdminReadonlyField(
+            admin_form, 'chapter', True, model_admin=related_admin
+        )
+
+        # Verify that a foreign key to the same app renders correctly
+        field_url = readonly_field.get_admin_url(Language.iso.field, language)
+        self.assertIn('/admin5/', field_url)
+
+        # Verify that a foriegn key to a different app renders correctly
```

Comment:
To add more explanation, comments should state the expected behavior and omit prefixes like "Verify that" since all tests verify things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).